### ISSUE TITLE
(1) Supprimer plusieurs champs d'un modèle en une fois

### ISSUE THREAD

**Issue by touv** [Inist-CNRS/lodex#2187]
### Discussed in https://github.com/Inist-CNRS/lodex/discussions/2061

<div type='discussions-op-text'>

<sup>Originally posted by **MathildeHuguin** June 24, 2024</sup>
**Constat**
Dans le cas de l'import d'un modèle existant dans une instance : il faut entrer individuellement dans chaque champ du modèle (que ce soit sur la page d'accueil, la ressource principale ou la page dédiée aux graphiques) pour les supprimer. 

**Idée d'amélioration**
Ajouter **(i)** une case pour pouvoir sélectionner tous les champs que l'on souhaite supprimer et **(ii)** un/des bouton(s) `supprimer tout` ou `supprimer uniquement la sélection` permettrait de gagner du temps lorsque l'on souhaite re-travailler sur un modèle existant.</div>

**Comment by Alexmarmelab** [comment]
**Solution**
Dans les pages Accueil, Ressources, Sous-ressources ( + clic sur une sous-ressource) et Graphique  : 
- ajouter une case à cocher sur chaque champ. La case à cocher doit être visible même lorsque le champ n'est pas survolé.
- Quand au moins une case est cochée, faire apparaître un bouton secondaire "Supprimer la sélection" 
- Au clic sur ce bouton, demander la confirmation de suppression : 
Titre : Supprimer la sélection
Description : Êtes-vous sûr de vouloir supprimer les XX champs sélectionnés ?
Liste : Etiquette [id] Icone(s) Nom interne 
Deux boutons : Annuler / Accepter


![Image](https://github.com/user-attachments/assets/9df8acbb-fdea-4a38-b56f-7ee0885a3c03)

![Image](https://github.com/user-attachments/assets/50af333f-1a5b-48ab-8d09-528061ff2fa1)

Exemple avec deux champs qui vont être supprimés :
<img width="903" alt="Image" src="https://github.com/user-attachments/assets/03f47003-7782-4091-b8c2-d2a8275b2409" />

**Comment by eonm-pro** [comment]
La fonctionnalité est OK, mais le texte de la maquette n'est pas respecté.

**Comment by AnaelKremer** [comment]
Ok pour moi aussi

**Comment by eonm-pro** [comment]
> Ok pour moi aussi

@AnaelKremer  je passe en to be fixed pour la correction du textet

**Comment by jonathanarnault** [comment]
Correction effectuée

**Comment by AnaelKremer** [comment]
~Le texte de la maquette n'est toujours pas le même, mais en même temps je ne sais pas à quoi tu faisais référence @eonm-pro ?
au fait que dans la maquette il était écrit le nombre de champs à supprimer ?~

~En tout cas ça me va tel quel~

**Comment by eonm-pro** [comment]
@AnaelKremer Je fais référence à cette capture d'écran envoyée par @Alexmarmelab plus haut dans la conversation ![](https://private-user-images.githubusercontent.com/130988888/396164231-03f47003-7782-4091-b8c2-d2a8275b2409.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzY3NzYyNTUsIm5iZiI6MTczNjc3NTk1NSwicGF0aCI6Ii8xMzA5ODg4ODgvMzk2MTY0MjMxLTAzZjQ3MDAzLTc3ODItNDA5MS1iOGMyLWQyYTgyNzViMjQwOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTEzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDExM1QxMzQ1NTVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05MjFjMjMxNjg5MzVhNzM2NzZjZDQyMjVjYmE5Nzc1YjQzMGFjNWY3Y2U3ZGRkZjIzZmQxOTBhMTNkZjk2MmZjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.LPuJx24mmOKyuDw7I88hd7i5CfEON3muIhN1FH1Y4bM)

**Comment by AnaelKremer** [comment]
non mais j'ai peut être testé trop vite après installation de la nouvelle version ! J'ai vu avec Jonathan, les modifs de toutes les PR n'étaient pas prises en comptes sur mon instance qui était bien en 74. Puis est passée en 73 dans ma barre de navigation...

Du coup là c'est ok 

![Image](https://github.com/user-attachments/assets/cd418f76-1d9b-47d1-b8df-338fe0173db9)

**Comment by eonm-pro** [comment]
Testé en 14.0.74 :

Ok pour moi.

**Comment by nantonot** [comment]
v14.0.75 : la suppression fonctionne
toutefois comme il y a dépublication/republication, un certain laps de temps est nécessaire pour voir la suppression effective

**Comment by Alexmarmelab** [comment]
Ajouter un loader sur le bouton

**Comment by jonathanarnault** [comment]
Loader ajouté sur le bouton de suppression.

Pour tester avec un petit dataset:
- Ouvrir les developer tools
- Onglet réseau
- Limiter le réseau au débit 3g (après avoir chargé la page)
- Lancer la suppression

**Comment by nantonot** [comment]
v14.0.76 : on voit bien que la suppression est en cours avec la roue de progression dans le bouton. Toutefois une fois la suppression terminée, le bouton n'apparait plus.

![Image](https://github.com/user-attachments/assets/8b930940-ae39-40cb-b5ed-e7d0100092e9)

**Comment by yannguillemin** [comment]
@nantonot : oui, il faut recliquer sur un ou des champs à supprimer pour le faire réapparaître.

**Comment by yannguillemin** [comment]
Sinon, je confirme aussi que cela fonctionne bien sur la V14.0.76.
